torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -59,10 +60,24 @@ std::map<at::ScalarType, ncclDataType_t> ncclDataType = {
 
 // Helper function that gets the data type and issues error if not supported
 ncclDataType_t getNcclDataType(at::ScalarType type) {
-  try {
-    return ncclDataType.at(type);
-  } catch (std::out_of_range& e) {
-    throw std::runtime_error("Unsupported data type for NCCL process group");
+  auto it = ncclDataType.find(type);
```

Comment:
Will it work if we add an entry to `ncclDataType` to map `at::kBool` to `ncclUint8`? How does `at::kBool` represent and interpret true and false? I recall had a discussion with @izdeby. IIRC, only 0 is interpreted as false and all others are true?

Reply:
Yes, the casting seems to be:

bool True -> 1, False -> 0
any nonzero -> True, 0 -> False

The issue I see with casting to `uint8` - either through the map or with similar logic as above - is that this will mean we can only support up to 255 processes calling `all_reduce()`. For example, if we have 256 processes and call allreduce with sum, and each process contributes a single set bit, we'd get 0 as our result, which would be wrong.